galsim/interpolatedimage.py

```diff
@@ -550,6 +546,182 @@ def __setstate__(self, d):
                       _force_stepk=self._stepk, _force_maxk=self._maxk)
 
 
+class InterpolatedKImage(GSObject):
+    """A class describing non-parametric profiles specified by samples -- i.e., images -- of their
+    complex Fourier transform.
+
+    The InterpolatedKImage class is useful if you have a non-parametric description of the Fourier
+    transform of an object as a pair of Images -- one for the real part and one for the imaginary
+    part -- of the complex-valued profile that you wish to manipulate / transform using GSObject
+    methods such as shear(), magnify(), shift(), etc.  Note that neither real-space convolution nor
+    photon-shooting of InterpolatedKImages is currently implemented.  Please submit an issue at
+    http://github.com/GalSim-developers/GalSim/issues if you require either of these use cases.
+
+    The primary intended use case for InterpolatedKImage is to create ChromaticObjects from
+    PSF-deconvolved multi-band HST images, a functionality that will be added in a future
+    pull-request.
+
+    The images required for creating an InterpolatedKImage are precisely those returned by the
+    GSObject `.drawKImage()` method.  The `a` and `b` objects in the following command will produce
+    identical images when drawn with the `.drawImage()` method:
+
+    >>> a = returns_a_GSObject()
+    >>> b = galsim.InterpolatedKImage(*a.drawKImage())
```

Comment:
This isn't really true.  You might say "...will produce essentially equivalent images", but identical is definitely overstating the case here.
```
>>> import galsim
>>> a = galsim.Gaussian(flux=1.7, sigma=2.3)
>>> b = galsim.InterpolatedKImage(*a.drawKImage())
galsim/interpolatedimage.py:674: UserWarning: Provided Real and Imag kimages do not meet GSParams.maxk_threshold criterion.
  "Provided Real and Imag kimages do not meet GSParams.maxk_threshold criterion.")
>>> im1 = a.drawImage(scale=0.25)
>>> im2 = b.drawImage(scale=0.25)
>>> im1.array
array([[  1.84585705e-18,   3.53282801e-18,   6.64980917e-18, ...,
          6.69098084e-18,   3.57130845e-18,   1.88142065e-18],
       [  3.57958274e-18,   6.77390280e-18,   1.26729025e-17, ...,
          1.26795224e-17,   6.77823682e-18,   3.58197536e-18],
       [  6.70457721e-18,   1.26956036e-17,   2.37427471e-17, ...,
          2.37967852e-17,   1.27480775e-17,   6.75501538e-18],
       ..., 
       [  6.71199826e-18,   1.26954282e-17,   2.37347450e-17, ...,
          2.37287711e-17,   1.26900094e-17,   6.70712369e-18],
       [  3.63048496e-18,   6.84009835e-18,   1.27541324e-17, ...,
          1.26999736e-17,   6.78783087e-18,   3.58060141e-18],
       [  1.89215476e-18,   3.57907113e-18,   6.69621442e-18, ...,
          6.68293279e-18,   3.56788310e-18,   1.88293336e-18]], dtype=float32)
>>> im2.array
array([[ -5.54597435e-08,  -5.58401254e-08,  -5.62082931e-08, ...,
         -6.89198032e-08,  -6.68243842e-08,  -6.48945928e-08],
       [ -3.36760735e-08,  -3.32897372e-08,  -3.28244560e-08, ...,
         -5.21154071e-08,  -4.99594144e-08,  -4.79943765e-08],
       [ -6.82893431e-09,  -5.73644288e-09,  -4.51138993e-09, ...,
         -2.74646297e-08,  -2.55707686e-08,  -2.38654927e-08],
       ..., 
       [ -2.38654927e-08,  -2.55707686e-08,  -2.74646297e-08, ...,
         -4.51138993e-09,  -5.73644288e-09,  -6.82893431e-09],
       [ -4.79943765e-08,  -4.99594144e-08,  -5.21154071e-08, ...,
         -3.28244560e-08,  -3.32897372e-08,  -3.36760735e-08],
       [ -6.48945928e-08,  -6.68243842e-08,  -6.89198032e-08, ...,
         -5.62082931e-08,  -5.58401254e-08,  -5.54597435e-08]], dtype=float32)
>>> numpy.max(im1.array)
0.0031840832
>>> numpy.max(im2.array)
0.0031777034
>>> im3 = a.drawImage(scale=0.25, method='fft')
>>> im3.array
array([[  1.84585705e-18,   3.53282801e-18,   6.64980917e-18, ...,
          6.69098084e-18,   3.57130845e-18,   1.88142065e-18],
       [  3.57958274e-18,   6.77390280e-18,   1.26729025e-17, ...,
          1.26795224e-17,   6.77823682e-18,   3.58197536e-18],
       [  6.70457721e-18,   1.26956036e-17,   2.37427471e-17, ...,
          2.37967852e-17,   1.27480775e-17,   6.75501538e-18],
       ..., 
       [  6.71199826e-18,   1.26954282e-17,   2.37347450e-17, ...,
          2.37287711e-17,   1.26900094e-17,   6.70712369e-18],
       [  3.63048496e-18,   6.84009835e-18,   1.27541324e-17, ...,
          1.26999736e-17,   6.78783087e-18,   3.58060141e-18],
       [  1.89215476e-18,   3.57907113e-18,   6.69621442e-18, ...,
          6.68293279e-18,   3.56788310e-18,   1.88293336e-18]], dtype=float32)
>>> numpy.max(im3.array)
0.0031840832
>>> numpy.max(im3.array) - numpy.max(im2.array)
6.3797925e-06
```
So with `method='fft'` the max is closer to equal, so that's better, but still numerically different.  So "identical" isn't really accurate here.